### PR TITLE
NumberMap

### PR DESCRIPTION
**Pull request by Rested**
All the tests work, not sure if I've set something up a bit wrong though.

### CODE REVIEW

File: stringmap.test.ts [hunk 1]
```diff
@@ -81,10 +82,93 @@ describe("stringmap", function() {
...
+        assert.isTrue(had, "delete didn't recognize previous inclusion");
+        assert.isFalse(map.has(5), 'had should be false after a delete');
+
+        const notPresent = map.delete(100);
+        assert.isFalse(notPresent, "map false positive on previous inclusion");
     })
 
 })
 
+describe("numbermap", function() {
```
Comment (timruffles) [inline stringmap.test.ts]: Did you want to pop the number map tests in here? Would also clean up some of the duplication in the test descriptions (i.e they all start with 'NumberMap')

File: stringmap.ts [hunk 2]
```diff
@@ -75,5 +75,82 @@ export class StringMap<V, K extends string = string> {
 }
 
 
+export class NumberMap<V, K extends number = number> {
```
Comment (timruffles) [inline stringmap.ts]: Shall we pop this in its own file?

File: target-typescript/stringmap.d.ts [hunk 3]
```diff
@@ -11,3 +11,16 @@ export declare class StringMap<V, K extends string = string> {
     keys(): K[];
     entries(): [K, V][];
 }
+export declare class NumberMap<V, K extends number = number> {
```
Comment (timruffles) [inline target-typescript/stringmap.d.ts]: `number` can't be usefully extended as far as I know, so I'd remove the generic from this type and just make `K` number.

**Review by timruffles — CHANGES_REQUESTED** [review]
Great start! I think there a few changes it makes sense to do? Let me know what you think.

**Review by timruffles — APPROVED** [review]
Great! LGTM

**Comment by Rested** [comment]
hmm think we still need to change the main variable in the package so that it's exporting both. How is it best to structure this? An index.ts file with this:
`import { StringMap } from "./stringmap";
import { NumberMap } from "./numbermap";

export const ES5Map = {
    NumberMap: NumberMap,
    StringMap: StringMap
}`

File: index.ts [hunk 4]
```diff
@@ -0,0 +1,3 @@
+export { StringMap } from "./stringmap";
```
Comment (Rested) [inline index.ts:1]: is this the proper way to go do this in ts?

Comment (timruffles) [reply index.ts:1]: That's great!

**Pull request merged**